handler/src/main/java/io/netty/handler/pcap/PcapWriter.java

```diff
@@ -58,22 +62,25 @@
      * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
      */
     void writePacket(ByteBuf packetHeaderBuf, ByteBuf packet) throws IOException {
-        long currentTime = System.currentTimeMillis();
+        long timestamp = System.currentTimeMillis();
 
         PcapHeaders.writePacketHeader(
                 packetHeaderBuf,
-                (int) TimeUnit.MILLISECONDS.toSeconds(currentTime),
-                (int) TimeUnit.MILLISECONDS.toMicros(currentTime) % 1000000,
+                (int) (timestamp / 1000L),
+                (int) (timestamp % 1000L * 1000L),
                 packet.readableBytes(),
                 packet.readableBytes()
         );
 
-        packetHeaderBuf.readBytes(outputStream, packetHeaderBuf.readableBytes());
-        packet.readBytes(outputStream, packet.readableBytes());
+        if (!isClosed) {
+            packetHeaderBuf.readBytes(outputStream, packetHeaderBuf.readableBytes());
+            packet.readBytes(outputStream, packet.readableBytes());
+        }
     }
 
     @Override
     public void close() throws IOException {
+        isClosed = true;
         outputStream.flush();
         outputStream.close();
```

Comment:
This method should ideally be idempotent, such that closing an already closed PcapWriter does nothing. Relying on the flush call on a closed OutputStream not throwing seems a bit shaky.